client-sdk/ts-web/ext-utils/src/protocol.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file Constants and type definitions for the `postMessage`-based protocol
+ * used between web content and the extension.
+ */
+
+// page <- message frame
+export const MESSAGE_TYPE_READY = 'oasis-xu-ready';
+export const MESSAGE_TYPE_RESPONSE = 'oasis-xu-response';
+
+export interface MessageReady {
+    type: typeof MESSAGE_TYPE_READY;
+}
+
+export interface MessageResponse {
+    type: typeof MESSAGE_TYPE_RESPONSE;
+    id: number;
+    body?: unknown;
+    err?: unknown;
+}
+
+// page -> message frame
+export const MESSAGE_TYPE_REQUEST = 'oasis-xu-request';
+
+export interface MessageRequest {
+    type: typeof MESSAGE_TYPE_REQUEST;
+    id: number;
+    body: unknown;
+}
+
+// methods
+
+export const METHOD_CONTEXT_SIGNER_PUBLIC = 'context-signer-public';
+
+export interface ContextSignerPublicRequest {
+    method: typeof METHOD_CONTEXT_SIGNER_PUBLIC;
+    /**
+     * An extra parameter, suggested to be used to help select which key to
+     * use. This lirbary passes it through verbatim; it's up to the extension
+     * to define how it's interpreted.
+     */
+    which: any;
```

Comment:
Could be useful to have a method for listing keys as well?

Reply:
o good idea

Reply:
ok I tried adding one

Reply:
Cool, looks good, thanks!